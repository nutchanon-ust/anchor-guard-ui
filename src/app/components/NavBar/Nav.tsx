import * as React from 'react';
import styled from 'styled-components/macro';
import { ReactComponent as DocumentationIcon } from './assets/documentation-icon.svg';
import { ReactComponent as GithubIcon } from './assets/github-icon.svg';

export function Nav() {
  return (
    <Wrapper>
      <Item
        href="https://cansahin.gitbook.io/react-boilerplate-cra-template/"
        target="_blank"
        title="Documentation Page"
        rel="noopener noreferrer"
      >
        <DocumentationIcon />
        Documentation
      </Item>
      <Item
        href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
        target="_blank"
        title="Github Page"
        rel="noopener noreferrer"
      >
        <GithubIcon />
        Github
      </Item>
    </Wrapper>
  );
}

const Wrapper = styled.nav`
  display: flex;
  margin-right: -1rem;
`;

const Item = styled.a`
  color: ${p => p.theme.primary};
  cursor: pointer;
  text-decoration: none;
  display: flex;
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  align-items: center;

  &:hover {
    opacity: 0.8;
  }

  &:active {
    opacity: 0.4;
  }

  .icon {
    margin-right: 0.25rem;
  }
`;
