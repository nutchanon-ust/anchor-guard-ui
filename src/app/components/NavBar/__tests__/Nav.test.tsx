import * as React from 'react';
import { render } from '@testing-library/react';
import { Nav } from '../Nav';
import { MemoryRouter } from 'react-router-dom';

describe('<Nav />', () => {
  it('should match the snapshot', () => {
    const logo = render(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>,
    );
    expect(logo.container.firstChild).toMatchSnapshot();
  });
});
