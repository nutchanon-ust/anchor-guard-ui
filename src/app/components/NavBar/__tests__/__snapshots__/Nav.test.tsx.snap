// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
}

.c1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:hover {
  opacity: 0.8;
}

.c1:active {
  opacity: 0.4;
}

.c1 .icon {
  margin-right: 0.25rem;
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="https://cansahin.gitbook.io/react-boilerplate-cra-template/"
    rel="noopener noreferrer"
    target="_blank"
    title="Documentation Page"
  >
    <svg>
      documentation-icon.svg
    </svg>
    Documentation
  </a>
  <a
    class="c1"
    href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
    rel="noopener noreferrer"
    target="_blank"
    title="Github Page"
  >
    <svg>
      github-icon.svg
    </svg>
    Github
  </a>
</nav>
`;
